ci-operator/step-registry/baremetal/lab/agent/day2/baremetal-lab-agent-day2-commands.sh

```diff
@@ -0,0 +1,181 @@
+#!/bin/bash
+
+set -o errtrace
+set -o pipefail
+set -o nounset
+
+# Trap to kill children processes
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM ERR
+
+
+[ -z "${AUX_HOST}" ] && { echo "\$AUX_HOST is not filled. Failing."; exit 1; }
+[ -z "${architecture}" ] && { echo "\$architecture is not filled. Failing."; exit 1; }
+[ -z "${workers}" ] && { echo "\$workers is not filled. Failing."; exit 1; }
+[ -z "${masters}" ] && { echo "\$masters is not filled. Failing."; exit 1; }
+
+if [ -f "${SHARED_DIR}/proxy-conf.sh" ] ; then
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+proxy="$(<"${CLUSTER_PROFILE_DIR}/proxy")"
+
+if [ "${ADDITIONAL_WORKERS}" == "0" ]; then
+    echo "No additional workers requested"
+    exit 0
+fi
+
+if [ "${ADDITIONAL_WORKERS_DAY2}" != "true" ]; then
+    echo "Skipping as the additional nodes have been provisioned at installation time."
+    exit 0
+fi
+
+function mount_virtual_media() {
+  local host="${1}"
+  local iso_path="${2}"
+  echo "Mounting the ISO image in #${host} via virtual media..."
+  timeout -s 9 10m ssh "${SSHOPTS[@]}" "root@${AUX_HOST}" mount.vmedia "${host}" "${iso_path}"
+  local ret=$?
+  if [ $ret -ne 0 ]; then
+    echo "Failed to mount the ISO image in #${host} via virtual media."
+    touch /tmp/virtual_media_mount_failure
+    return 1
+  fi
+  return 0
+}
+
+function get_ready_nodes_count() {
+  oc get nodes \
+    -o jsonpath='{range .items[*]}{.metadata.name}{","}{.status.conditions[?(@.type=="Ready")].status}{"\n"}{end}' | \
+    grep -c -E ",True$"
+}
+
+function approve_csrs() {
+  while [[ ! -f '/tmp/agent-day2-complete' ]]; do
+    echo "approve_csrs() running..."
+    oc get csr -o go-template='{{range .items}}{{if not .status}}{{.metadata.name}}{{"\n"}}{{end}}{{end}}' \
+      | xargs --no-run-if-empty oc adm certificate approve || true
+    sleep 30
+  done
+}
+
+function monitor_workers()
+{
+  local DAY2_IPS=""
+  expected_nodes=$(( $(get_ready_nodes_count) + ADDITIONAL_WORKERS ))
+  # shellcheck disable=SC2154
+  for bmhost in $(yq e -o=j -I=0 '.[] | select(.name|test("-a-"))' "${SHARED_DIR}/hosts.yaml"); do
+    # shellcheck disable=SC1090
+    . <(echo "$bmhost" | yq e 'to_entries | .[] | (.key + "=\"" + .value + "\"")')
+    DAY2_IPS+="${ip},"
+  done
+  DAY2_IPS=${DAY2_IPS%,}
+  /tmp/oc adm node-image monitor --ip-addresses "${DAY2_IPS}" -a "${day2_pull_secret}" --insecure=true
+  ready_nodes=$(get_ready_nodes_count)
+  if [ "${ready_nodes}" != "${expected_nodes}" ]; then
+    echo "Day2 install fail..."
+    exit 1
+  fi
+  touch /tmp/agent-day2-complete
+  return 0
+}
```

Comment:
This function can be removed as per suggestion here https://github.com/openshift/release/pull/55608#discussion_r1731158760